### ISSUE TITLE
[Feature Request] - Make all warnings errors

### ISSUE THREAD

**Issue by C-Otto** [google/error-prone#1589]
I'd like to fail my build if Error Prone finds a warning in my code. Then, I'm forced to fix the code, or suppress the warning. I'm aware of false positives, and I'm happy to take the risk.

Please have a look at #614 for more information (feel free to re-open and continue there).

**Comment by tbroyer** [comment]
Do you use `-Werror`? Works great for me.

**Comment by C-Otto** [comment]
Thanks, didn't know about that (see Twitter).

**Comment by C-Otto** [comment]
```
tasks.withType(JavaCompile) {
    options.compilerArgs << "-Werror"
}
```
or
```
compileJava {
        options.compilerArgs << "-Werror"
}
```
